src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,45 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    if (tag != null && !tag.isEmpty() && matchPattern.startsWith(tag)) {
+      return matchPattern.substring(tag.length()).equals("*");
```

Comment:
```
127.0.0.1:7379> set {hello}world1 1
OK
127.0.0.1:7379> set {hello}world2 1
OK
127.0.0.1:7379> set {hello}world3 1
OK
127.0.0.1:7379> set {hello}world4 1
OK
127.0.0.1:7379> set {hello}man1 1
OK
127.0.0.1:7379> set {hello}man2 1
OK
127.0.0.1:7379> scan 0 MATCH {hello}world*
1) "0"
2) 1) "{hello}world1"
   2) "{hello}world3"
   3) "{hello}world4"
   4) "{hello}world2"
127.0.0.1:7379> scan 0 MATCH {hello}man*
1) "0"
2) 1) "{hello}man1"
   2) "{hello}man2"
127.0.0.1:7379> scan 0 MATCH {hello}man1
1) "0"
2) 1) "{hello}man1"
```

Please refer above result. The thing is, we shouldn't restrict to use only '*'. It's just a one of use case.